### ISSUE TITLE
If Microsoft EDGE is still required, then the Bing protocol has not yet been reverse engineered?

### ISSUE THREAD

**Issue by Emasoft** [acheong08/EdgeGPT#84]
If Microsoft EDGE is still required, then the Bing protocol has not yet being reverse engineered.
Can you get rid of the EDGE requirement? So that we can run it on a Linux machine for example.

**Comment by scmanjarrez** [comment]
You don't need Edge, only to retrieve the cookies in order to authenticate with the endpoints. [acheong08](https://github.com/acheong08) has built this library on top of these endpoints.

**Comment by Ataraksia** [comment]
Also, just as a heads up, you can absolutely run Edge on Linux.  It's just another Chromium reskin after all.  Here's the stable branch on the AUR for example: https://aur.archlinux.org/packages/microsoft-edge-stable-bin

Works just fine for the purposes of obtaining the required cookie.

**Comment by Emasoft** [comment]
I was talking about headless linux machines.

**Comment by scmanjarrez** [comment]
As I said, you don't need it at all. Retrieve the cookies from another computer and copy them to the machine running EdgeGPT. I have a telegram bot running EdgeGPT in a headless server, I only copied the cookies file there.

**Comment by Ataraksia** [comment]
And just to follow-up on this, I actually didn't read the instructions that closely the first time as the whole process went so smoothly (on a non headless server mind you) that I didn't need to, but looking back I can't see any reason Edge would be necessary at all?  I just checked to make sure and I was able to get a working cookie from Chrome and Firefox just the same as Edge, and in fact now that I'm looking more closely, the instructions themselves even say get the cookie with Firefox or Chrome;  Edge is only mentioned for verification purposes that you actually have access to BingChat in the first place, which I can only assume is there in an attempt by  acheong08 to try and cut down on some of the inevitable "why isn't this working" comments from people who never had access to BingChat in the first place.  So yeah, get the cookie from any Chromium or Firefox based browser that strikes your fancy on the headed computer of your choice and just copy it too your headless setup.  Easy peasy.
